pyspecProcScripts/correlation_alignment.py

```diff
@@ -77,64 +79,29 @@ def correl_align(
-    #         *always* want to align along nScans
-    #         (so we would smoosh the indirect
-    #         dimension and `nScans` to make
-    #         `repeat_dim`).
-    #         Most likely, we want to specify a
-    #         list of dimensions that are "safe"
-    #         to smoosh together as `repeat_dim`,
-    #         assuming that `nScans` is *always*
-    #         safe.
-    #         I think that right now, what you
-    #         call `indirect_dim` will be part of
-    #         `repeat_dims`.
-    # TODO ☐: Furthermore, we want to specify
-    #         `non_repeat_dims`.  These are also
-    #         indirect dimension, but they are
-    #         one that we are not OK to align
-    #         (e.g. the indirect dimension of a 2D
-    #         COSY experiment).
-    #         With both `repeat_dims` and
-    #         `non_repeat_dims` defined, we can use a set
-    #         expression like below to find the
-    #         indirect dimension (not phase
-    #         cycling and not direct) and to make
-    #         sure that they are all either (1)
-    #         nScans (2) specified as safe or (3)
-    #         listed in `non_repeat_dims`
-    #         Currently, I don't think you have
-    #         any `non_repeat_dims`
-    if avg_dim:
-        phcycdims = [j for j in s_orig.dimlabels if j.startswith("ph")]
-        indirect = set(s_orig.dimlabels) - set(phcycdims) - set([direct])
-        # TODO ☐: see todo above -- you need to
-        #         first make repeat_dims and
-        #         non_repeat_dims lists of length
-        #         1 if they are given as strings
-        # assert len(indirect - set(repeat_dims) - set(non_repeat_dims)) = 0
-        indirect = [j for j in s_orig.dimlabels if j in indirect]
-        s_jk = s_orig.C.smoosh(indirect)  # this version ends up with
+    assert indirect_dim is None, (
+        "We updated the correlation function to no longer take indirect_dim as"
+        " a kwarg!"
+    )
+    assert avg_dim is None, (
+        "We updated the correlation function to no longer take avg_dim as a"
+        " kwarg!"
+    )
+    phcycdims = [j for j in s_orig.dimlabels if j.startswith("ph")]
+    indirect = set(s_orig.dimlabels) - set(phcycdims) - set([direct])
+    assert len(indirect - set(repeat_dims) - set(non_repeat_dims)) == 0
+    indirect = [j for j in s_orig.dimlabels if j in indirect]
+    if len(indirect) > 1:
+        s_jk = s_orig.C.smoosh(
+            indirect, "repeats"
+        )  # this version ends up with
         #                                   three dimensions
         #                                   (j=align_dim, k=phcyc, and
         #                                   direct nu) and is NOT conj
     else:
-        s_jk = s_orig.C  # even if there isn't an indirect to smoosh we will
+        s_jk = s_orig.C.rename(
+            indirect[0], "repeats"
+        )  # even if there isn't an indirect to smoosh we will
```

Comment:
to be consistent with the above (when indirect length is more than 1) I just rename so I can keep the name throughout the rest of the code

Reply:
this is good, but again:

1.  you mean `repeat_dims` not `indirect`
2. have a check above if `type(repeat_dims) is str` (then make a list of length 1)
3. add `nScans` (if there is an nScans dimension!!!)
4. after all that, but before the chunk, throw an error if there is not `repeat_dims`

Reply:
- 2 is simplified to just checking it's a list and if not it says it needs to be a list of strings

- addressed 3 around lines 112

- 4 is now incorporated